### ISSUE TITLE
White flash when opening a new window maximized or fullscreen

### ISSUE THREAD

**Issue by RobbyCBennett** [microsoft/vscode#132339]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.60.0 (user setup)
Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
Date: 2021-09-01T10:41:52.311Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043

Steps to Reproduce:

1. Set "Window: New Window Dimensions" to "maximized" or "fullscreen"
2. From an existing instance of vscode, open a new instance

The issue was reported [here](https://github.com/microsoft/vscode/issues/95190) and closed because Electron 8 apparently fixed the issue. However, I'm still having this issue with Electron 13 on Windows. The issue is not present on Manjaro.

[Workaround](https://github.com/microsoft/vscode/issues/132339#issuecomment-3092420935)

**Comment by RobbyCBennett** [comment]
![78008708-56a84600-7383-11ea-8c60-90d608a89e56](https://user-images.githubusercontent.com/22121365/132599092-ccb0992d-c9f4-4e76-bd32-9fbeeeedf10e.gif)

**Comment by deepak1556** [comment]
Are you running via WSL ? Does the issue repro if you launch with `code --disable-gpu` ?

**Comment by RobbyCBennett** [comment]
No, I'm not using WSL to launch. That is just a gif I found on a previous issue that was similar, but that issue had a different cause and fix. The white flash looks the exact same to me. It occurs when I launch from the Windows GUI, from CMD, or from WSL.

I only have a dedicated graphics card and no integrated graphics on my Ryzen CPU, so I wouldn't expect anything different to happen. I have tried `code --disable-gpu` in various CLIs and I still have the white flash.

**Comment by deepak1556** [comment]
Can you provide a screen capture from your device showing the exact steps of launch ? Also attach the output of `code --verbose`

**Comment by RobbyCBennett** [comment]
Like I said, it happens when launching from the terminal or from the GUI. It looks exactly like the GIF, and the flash of white is the exact same. I could record 6 different gifs, one from the launching in taskbar, one from the start menu, one from a desktop icon, from CMD, Powershell, and WSL. It all results in the same thing.

`
[main 2021-09-09T03:32:20.601Z] Sending some foreground love to the running instance: 3180
[main 2021-09-09T03:32:20.606Z] Sending env to running instance...
[main 2021-09-09T03:32:20.728Z] Sent env to running instance. Terminating...
[main 2021-09-09T03:32:20.728Z] Lifecycle#kill()
`

**Comment by deepak1556** [comment]
> I could record 7 different gifs, one from the launching in taskbar, one from the start menu, one from a desktop icon, a new window from VSCode, from CMD, Powershell, and WSL. 

Yes it is fine, please attach a recording of what you are observing. 

As for the `code --verbose` output, looks like you are already having a running instance. Please close all running instances and run that command.

**Comment by RobbyCBennett** [comment]
Here's the log file, although it's a bit cluttered because of extensions and workspaces. I have already reproduced the error without extensions on a new install of VSCode Insiders, so extensions, themes, and workspaces are irrelevant.
[output.txt](https://github.com/microsoft/vscode/files/7133272/output.txt)

**Comment by RobbyCBennett** [comment]
Were you able to reproduce the issue? I couldn't get the screencast website you shared to work. Anyway, this makes no difference.

**Comment by yume-chan** [comment]
I can reproduce this

```
Version: 1.63.0-insider (user setup)
Commit: d18d093403b12a65350c58a7b0d5771cc1f42aba
Date: 2021-11-26T08:21:30.810Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043
```

[log.txt](https://github.com/microsoft/vscode/files/7616749/log.txt)

https://user-images.githubusercontent.com/1330321/143835118-5232eecc-d285-446c-a2e1-3563f63e71fc.mp4

**Comment by lllopo** [comment]
I've started getting this recently. Like 3-4 builds ago along with another more annoying issue - when VS code touches the bottom of the screen (even not maximized) it prevents the taskbar from showing up when I move the mouse to the bottom.

**Comment by rzhao271** [comment]
Upstream issue:
https://github.com/electron/electron/issues/35362

Chromium gets the message handler to show a window before the content window. The window that the message handler shows has a white background.

I notice that if I edit Chromium's OnPaint handler, I can get it to paint a black background by moving [a line that calls `FillRect`](https://source.chromium.org/chromium/chromium/src/+/main:ui/views/win/hwnd_message_handler.cc;l=2538) out of the if statement so that it always runs. This is more a proof of concept rather than an actual fix for the issue.

However, even with that fix, there is still a flash of white at the beginning.
That flash might be a [Windows bug](https://stackoverflow.com/questions/69715610/how-to-initialize-the-background-color-of-win32-app-to-something-other-than-whit). Therefore, assuming I get to merge in a fix into Electron for this issue, the fix would shorten the length of the white background, rather than completely eliminate it.

**Comment by rzhao271** [comment]
The bug still persists on Windows 11

**Comment by RobbyCBennett** [comment]
I have noticed that the issue is not currently present on Windows 11, but it is still present on Windows 10. I have made sure that VS Code is updated on both machines.

**Comment by lllopo** [comment]
Yes, definitely still present on Win 10 / latest insiders. Ms Edge and Chrome suffer from the same issue, btw.

**Comment by RobbyCBennett** [comment]
On Windows 11, I'm seeing the issue again for the latest release.

Version: 1.77.0 (system setup)
Commit: 7f329fe6c66b0f86ae1574c2911b681ad5a45d63
Date: 2023-03-29T10:02:16.981Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: No

**Comment by simon-20** [comment]
I also see this issue using Ubuntu 22.04.3 LTS.

It happens when I run `code --disable-gpu --disable-extensions`

As further information, it displays this white flash whenever I switch to the desktop workspace on which VS Code is running.

Happy to provide further information, if someone tells me what is needed and how to provide it.

If anyone knows a temporary work around for this, of any sort, that would be much appreciated, because having it flash every time one switches to the window is really frustrating.

**Comment by RobbyCBennett** [comment]
I've just tried this on a Windows 10 system that has an integrated GPU and a dedicated GPU and this time I noticed a difference between the two. Here are my results.

* Dedicated GPU (`code --disable-extensions`): the white flash is longer.
* Integrated GPU (`code --disable-gpu --disable-extensions`): the white flash is shorter. The 2nd time out of 3 times I ran this command, I didn't even notice a white flash.

https://github.com/microsoft/vscode/assets/22121365/2b0c2ac4-e00c-4aa5-a8a1-b471b4e2ffe8

**Comment by deepak1556** [comment]
Based on https://github.com/microsoft/vscode/issues/132339#issuecomment-1246051350 and given we don't have a clear path to upstream the changes to chromium, closing as not actionable.

**Comment by adityaonx** [comment]
I found a fix or say workaround for this white flash issue when opening maximized VS Code, had to experiment with various values.
I found that VS Code makes a delay on purpose when vscode is maximized or in fullscreen mode. This is to wait until vscode window is fully loaded and then show it. While waiting, it makes default white flash which is an electron default mechanism.
White Flash Fix:
1. Open C:\Users\<UserName>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\main.js
2. Search for "show:"
3. Replace line "show:e.mode!==0&&e.mode!==3" with "show:true,".
4. Save & Exit.
This will remove the delay and instantly loads the window which avoids white flash of electron.

**Comment by bpasero** [comment]
@deepak1556 any idea why `show: false` gives us white background flicker while `show: true` does not?

**Comment by deepak1556** [comment]
We know from previous investigation that the white background is from the window drawn by the message handler https://source.chromium.org/chromium/chromium/src/+/main:ui/gfx/win/window_impl.cc;l=218-220 , this window is not hidden in both show: false and show: true cases. It could only be that when drawing content window together with this message handling window it gets put behind the content window and we don't see the flash. We can attempt to hide the message handling window initially and see how that works.

**Comment by bpasero** [comment]
@deepak1556 so this is a different window than the one VS Code gets to be rendered in and with `show: false` it becomes initially briefly visible, while with `show: true` our window wins?

**Comment by deepak1556** [comment]
yup that is correct

**Comment by Frulfump** [comment]
I still get the white flash on https://github.com/microsoft/vscode/commit/496ebc4723371f29c9ffa0319dcccb2d7bee7ee0

**Comment by bpasero** [comment]
Can you try to run `code-insiders --disable-extensions --user-data-dir <directory>` where `<directory>` is an empty folder? This will ensure Code is starting with a fresh data directory, e.g. no specific settings and without any extensions running.

**Comment by Frulfump** [comment]
I tried it now and it's still the same.

**Comment by adityaonx** [comment]
> I tried it now and it's still the same.

Could you try on Windows 11 please. I think Windows 10 doesn't have the mica and various backdrop leaving white window in bg! if it still exists even in windows 11, then this must be an issue! Alternatively, for now you can try this workaround: https://github.com/microsoft/vscode/issues/132339#issuecomment-2766534068

**Comment by robbycbennett** [comment]
I see the white flash on Windows 11 with ~fullscreen~ the window maximized and the command line arguments from https://github.com/microsoft/vscode/issues/132339#issuecomment-2897906464.

```
Version: 1.101.0-insider
Commit: 7197360b79e6da3404ed5ad1fb4d7ea74a8bb5d2
Date: 2025-06-04T05:03:51.910Z
Electron: 35.5.0
ElectronBuildId: 11678209
Chromium: 134.0.6998.205
Node.js: 22.15.1
V8: 13.4.114.21-electron.0
OS: Windows_NT x64 10.0.26100
```

**Comment by robbycbennett** [comment]
Here's a screen recording. I had to open the window, maximize it, close it, then open it again to see the white flash. When it's a small non-maximized window, it doesn't flash white.

https://github.com/user-attachments/assets/a8d8e4f2-a520-4dda-92fe-83ad8742294b

**Comment by adityaonx** [comment]
> Here's a screen recording. I had to open the window, maximize it, close it, then open it again to see the white flash. When it's a small non-maximized window, it doesn't flash white.
> 
>  2025-06-04.16-52-17.mp4

It's strange how this bug was marked fixed and closed even without testing! Is this how they are handling tickets now? Sad state! @deepak1556 @bpasero

**Comment by Frulfump** [comment]
I saw that the milestone was changed to June but I still want to confirm it's still happening with the latest release from ~1 hour ago

Version: 1.102.0-insider (user setup)
Commit: 7e4e0f4e55d0d0a2a931aa4e0b7acc518e5da0dd
Date: 2025-06-13T05:04:03.051Z
Electron: 35.5.1
ElectronBuildId: 11727614
Chromium: 134.0.6998.205
Node.js: 22.15.1
V8: 13.4.114.21-electron.0
OS: Windows_NT x64 10.0.19045

**Comment by LinhTrucVo** [comment]
I updated to the latest version but the issue is still there

![Image](https://github.com/user-attachments/assets/128ab6ef-1943-4b69-9572-9b3f432b31de)

https://github.com/user-attachments/assets/1c5f397e-6e47-408e-84b0-baca3fcfa856

**Comment by BowkC** [comment]
I am also experiencing this issue
Brand new laptop Win 11
Vers 1.101.2

**Comment by robbycbennett** [comment]
Here's a python script that makes the workaround easier.

```py
import os

OLD = 'show:e.mode!==0&&e.mode!==3'
NEW = 'show:true'

PATHS = [
	f'{os.environ['SystemDrive']}/Program Files/Microsoft VS Code/resources/app/out/main.js',
	f'{os.environ['LocalAppData']}/Programs/Microsoft VS Code/resources/app/out/main.js',
]

for path in PATHS:
	try:
		with open(path, 'r') as file:
			data = file.read()
	except FileNotFoundError:
		continue

	data = data.replace(OLD, NEW, 1)

	with open(path, 'w') as file:
		file.write(data)

	print(f'Fixed {path}')
```

**Comment by adityaonx** [comment]
> I found a fix or say workaround for this white flash issue when opening maximized VS Code, had to experiment with various values. I found that VS Code makes a delay on purpose when vscode is maximized or in fullscreen mode. This is to wait until vscode window is fully loaded and then show it. While waiting, it makes default white flash which is an electron default mechanism. White Flash Fix:
> 
> 1. Open C:\Users<UserName>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\main.js
> 2. Search for "show:"
> 3. Replace line "show:e.mode!==0&&e.mode!==3" with "show:true,".
> 4. Save & Exit.
>    This will remove the delay and instantly loads the window which avoids white flash of electron.

I have already posted the solution previously. Why creating a duplicate comment each time with the same workaround?

**Comment by SkyOps117** [comment]
@adityaonx Sorry I didn't see it.
Is main.ts in the source code editable to fix that?

**Comment by robbycbennett** [comment]
> # No need for python script
> > Here's a python script that makes the workaround easier.
> 
> [@RobbyCBennett](https://github.com/RobbyCBennett)
> 
> The file to edit is plain text. Notepad, VSCode or any text editor will do. No more than a minute or two to edit manually.

Every time the editor updates, the main.js file changes. My script is just a way to automate the workaround. You're free to do it manually if you prefer.

**Comment by Frulfump** [comment]
This is labeled with "upstream-issue-fixed" which is incorrect https://github.com/electron/electron/issues/35362 is still open.

I added the basic info to that issue but might as well include it here

Still an issue, repros with Electron 37.5.1 in VS Code Insiders 1.105.0-insider (not able to check stable atm)

Version: 1.105.0-insider (user setup)
Commit: 6e60e679fc6b6a512edbed9a750d7c5811595861
Date: 2025-10-01T05:02:28.879Z
Electron: 37.5.1
ElectronBuildId: 12409799
Chromium: 138.0.7204.251
Node.js: 22.19.0
V8: 13.8.258.32-electron.0
OS: Windows_NT x64 10.0.19045

**Comment by adityaonx** [comment]
https://github.com/microsoft/vscode/pull/281765
No Updates to this PR from 2 days ago!

**Comment by Frulfump** [comment]
This no longer repros for me but I'm now on Windows 11 (25H2) as opposed to Windows 10 for the previous reported instances from me.

```
Version: 1.107.0 (user setup)
Commit: 618725e67565b290ba4da6fe2d29f8fa1d4e3622
Date: 2025-12-10T07:43:47.883Z
Electron: 39.2.3
ElectronBuildId: 12895514
Chromium: 142.0.7444.175
Node.js: 22.21.1
V8: 14.2.231.21-electron.0
OS: Windows_NT x64 10.0.26200
```
